### DISCUSSION TITLE
Sky map from spacecraft in deep space

### DISCUSSION THREAD

**Discussion by FabrizioPinto** [Stellarium/stellarium#2468] [Ideas]
Obviously Stellarium is very sophisticated in its ability to create a representation of the sky from many pre-assigned locations. Is it possible to create a sky chart as seen from a spacecraft with assigned coordinates? Maybe X, Y, Z in a ecliptic system or equivalent. I do not see that from the GUI. My students would love to see the sky as seen from spacecraft whose trajectory we may know from other sources (maybe JPL, or the like). Apologies if it is already possible by scripting. It would seem to be a capability within reach given the flexibility to map the sky from so many objects. If so, I will appreciate a link to those instructions. Great job! Stellarium is an amazing teaching tool for me. Fabrizio

**Comment by gzotti** [comment]
It is not easy currently. You could create an object on a Keplerian orbit around the sun and observe from there. See the "Tesla Roadster (Starman)" object. 

Free navigation in the Solar System would have to be developed, but is not the current focus of development. There are only so many weekends...

**Reply by FabrizioPinto** [reply]
Hello Georg, Thanks for the quick reply. It is always great to know the state of the art. Of course, at NASA (I worked at JPL in the previous century), we had in-house tools to plan on observations from spacecraft, Cassini (for instance). Similarly the HST or the JWST need such a tool. However, I struggle to find one in the FOSS world. You correctly state that one could define a Keplerian trajectory but that rules out more realistic computations including radiation pressure, drag, and N-body perturbations. If one could upload X(t) etc as a text file obtained from previously calculations with other tools, Stellarium could do its unparalleled  magic. I thought it was worth a shot. Good luck and I hope a weekend for this comes, eventually! Fabrizio

**Reply by gzotti** [reply]
I know there are trajectories from spacecraft available. (SPICE/NAIF data etc.). Just that somebody has to write some code, extending the Orbit class to one that reads such file, or even allow interactive control (add propulsion to change Keplerian trajectory). It's not even very difficult. Anybody could do it. Nobody has done so far... I have created a different simulator years ago, showing the (original) Rosetta trajectory and "instantaneous Kepler orbits". Very instructive. But I leave trajectory planning to JPL ;-)

**Reply by FabrizioPinto** [reply]
To be clear, what is the language in which this should be done that nobody has done? I am an old Fortran dog (I do not mean 77). If one wrote that in Modern Fortran (95 or whatever), could that be wrapped in the language Stellarium uses for plug-ins? Forgive the naivete, I am not a software designer and much less an OOP developer. But I write my own code as needed for research/teaching. Is there a recipe for a deliverable of this type? FP

**Reply by alex-w** [reply]
We use C++/Qt
